### PR TITLE
Added org todos and agenda to home screen

### PR DESCRIPTION
**Pull request by tonyday567**
replacing #3720

### CODE REVIEW

**Comment by robbyoconnor** [comment]
:+1:

File: core/core-spacemacs-buffer.el [hunk 1]
```diff
@@ -540,6 +540,125 @@ HPADDING is the horizontal spacing betwee the content line and the frame border.
...
+              (progn
+                (setq marker (get-text-property 0 'org-marker todo))
+                (setq pos (marker-position marker))
+                (setq txt (get-text-property 0 'txt todo))
+                (setq time (get-text-property 0 'time todo))
+                (list (cons "text" txt)
+                      (cons "file" file)
+                      (cons "pos" pos)
+                      (cons "time" time)))
+              ) rtn)))))
```
Comment (syl20bnr) [inline core/core-spacemacs-buffer.el]: Please can you break this into sub-functions :-)

File: core/core-spacemacs-buffer.el [hunk 2]
```diff
@@ -540,6 +540,125 @@ HPADDING is the horizontal spacing betwee the content line and the frame border.
...
+             (todo)
+              (progn
+                (setq marker (get-text-property 0 'org-marker todo))
+                (setq pos (marker-position marker))
+                (setq txt (get-text-property 0 'txt todo))
+                (setq time (get-text-property 0 'time todo))
+                (list (cons "text" txt)
+                      (cons "file" file)
+                      (cons "pos" pos)
+                      (cons "time" time)))
```
Comment (syl20bnr) [inline core/core-spacemacs-buffer.el]: This code seems to be duplicated, one more reason for a function.

**Comment by tonyday567** [comment]
@syl20bnr refactored.  I'm not very experienced with elisp - I used a 'loop for collect' but wasn't sure if this is a common idiom.  If there's any other improvements or tips, happy to incorporate.

**Comment by syl20bnr** [comment]
Thank you :+1: 
There some rough edges with it:
- the point jumps to EOF, it is caused by `(require 'org-agenda)`, I did not dig further
- There are some errors with `org-bullets-mode` but it may be specific with my configuration

Cherry-picked into develop branch, you can safely delete your branch.

**Comment by Andre0991** [comment]
@syl20bnr The `org-bullets-mode` thing is an upstream problem: https://github.com/syl20bnr/spacemacs/issues/4793

I don't know if the fixed version is already in the repo.

**Comment by fintelkai** [comment]
I like the idea of this but many of my todos and agenda items have links (to emails in particular) in them and since the *spacemacs* home buffer is not an org buffer, the links aren't fontified the way they would be in an org-mode buffer. That kind of makes this ugly and not so useful.

**Pull request closed without merging**